src/createMollieClient.ts

```diff
@@ -99,9 +99,18 @@ function createHttpClient({ apiKey, accessToken, versionStrings, ...axiosOptions
   axiosOptions.headers['Content-Type'] = 'application/json';
 
   axiosOptions.httpsAgent = new https.Agent({
-    cert: fs.readFileSync(path.resolve(__dirname, './cacert.pem'), 'utf8'),
+    ca: caCertificates,
   });
 
+  // Detect stub polyfills of https.Agent. Using a polyfill (such as https://github.com/jhiesey/stream-http) in itself
+  // is not an issue, although it may not respect the certificates and thereby reduce security somewhat. The purpose of
+  // this check is to catch cases where this library is integrated into a frontend app.
+  if (axiosOptions.httpsAgent?.requests == undefined) {
```

Comment:
Why not check for the presence of the `window` object?

Reply:
Another alternative would be checking `require('process').release.name`<sup>[1](https://nodejs.org/api/process.html#process_process_release)</sup>. I would prefer doing that over checking `window`. Polyfilling `window` is common (React Native does it, to name one).

I don't think there's a perfect solution. Ideally, we would check whether the library is running on a user's machine. But even Node.js could be running on a user's machine (think Electron).

Should we check `release.name` instead of `httpsAgent.requests`?

Reply:
Valid point on the polyfills – didn't think of that.

Let's go for `release.name` 👍